### PR TITLE
Remove deprecated 'app' parameter.

### PR DESCRIPTION
**Pull request by tomchristie**
This pull request deals with removing the already deprecated `app=...` parameter, which has been dropped in favor of the more explicit `transport = httpx.WSGITransport(...)` and `transport = httpx.ASGITransport()` configurations.

Related: #3314

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+## Unreleased
+
+* The deprecated `proxies` argument has now been removed.
```
Comment (karpetrosyan) [inline CHANGELOG.md:9]: Is this changelog related to the current PR?

**Review by zanieb — APPROVED** [review]

**Comment by tomchristie** [comment]
Thanks @zanieb ☺️
